### PR TITLE
#3690 adding RequestURL, Method, Operation and Payload fields to APIResponse

### PR DESCRIPTION
**Pull request by neilotoole**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run`./bin/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Addresses #3690 

In case it isn't clear in #3690, the motivation for making `RequestURL`, `Method` etc available on the `APIResponse` object, is that sometimes the embedded `*http.Response` can be `nil`, therefore the caller cannot access this information.

Filed against `master` as it does not contain breaking changes.

### CODE REVIEW

**Comment by ePaul** [comment]
@neilotoole I don't really have any comments about the PR's content, but could you mention in the initial comment that it is for the language Go? (I.e. replace `{LANG}` with `Go` where applicable.) (Possibly also mention it in the title.) This makes it easier for me to filter out the "uninteresting" Pull requests.

File: modules/swagger-codegen/src/main/resources/go/api.mustache [hunk 1]
```diff
@@ -135,11 +136,20 @@ func (a {{{classname}}}) {{{nickname}}}({{#allParams}}{{paramName}} {{{dataType}
...
+
+	var localVarURL, _ = url.Parse(localVarPath)
+	localVarURL.RawQuery = localVarQueryParams.Encode()
+	var localVarAPIResponse = &APIResponse{Operation: "{{operationId}}", Method: localVarHttpMethod, RequestURL: localVarURL.String()}
+	if localVarHttpResponse != nil {
+		localVarAPIResponse.Response = localVarHttpResponse.RawResponse
+		localVarAPIResponse.Payload = localVarHttpResponse.Body()
+	}
+
+if err != nil {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/go/api.mustache]: @neilotoole this line is not indented properly.

**Comment by wing328** [comment]
Please rebase on the latest master to resolve the merge conflicts.

cc @guohuang

**Comment by wing328** [comment]
I've made some minor fixes to indentation and merged the changes into the master.

**Pull request merged**